zubhub_frontend/zubhub/src/views/project_details/ProjectDetails.jsx

```diff
@@ -324,14 +327,14 @@ function ProjectDetails(props) {
                   <Grid item xs={12} sm={12} md={12} align="center">
                     <Box className={classes.sliderBoxStyle}>
                       <Slider {...sliderSettings(project.images.length)}>
-                        {project.images.map(image => (
+                        {project.images.map((image, index) => (
                           <div>
                             <img
                               key={image.public_id}
                               className={classes.carouselImageStyle}
                               src={image.image_url}
                               alt={image.public_id}
-                              onClick={e => handleSetState(handleOpenEnlargedImageDialog(e, state))}
+                              onClick={e => handleSetState(handleOpenEnlargedImageDialog(e, state, index))}
```

Comment:
Its better to use either the image ref or image id rather than index. In cases like slicing array or infinite scrolling it might affect indices.